### PR TITLE
EPMDJ-5058: integration with new sort/filter realization

### PR DESCRIPTION
**Pull request by IliaSolovev**
Issue: EPMDJ-5058, EPMDJ-5032

### CODE REVIEW

File: src/modules/table-actions/table-actions-context.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
-import { TableActionsState, Order } from './table-actions-types';
+import { TableActionsState } from './table-actions-types';
 import { Action } from './reducer';
 
-export const defaultState = {
-  search: { fieldName: 'name', value: '' },
-  sort: { fieldName: 'name', order: 'ASC' as Order },
+export const defaultState: TableActionsState = {
+  search: [{ field: 'name', value: '', op: 'CONTAINS' }],
```
Comment (pavyarov) [inline src/modules/table-actions/table-actions-context.ts:7]: op has a default value at BE?

File: src/pages/agent-full-page/coverage-plugin/coverage-details/coverage-details.tsx [hunk 2]
```diff
@@ -56,8 +56,8 @@ export const CoverageDetails = coverageDetails(
                 </form>
               )}
             />
-            <SearchResult align={search.value ? 'space-between' : 'end'}>
-              {search.value && <span data-test="coverage-details:search-result">{coverageByPackages.length} result</span>}
+            <SearchResult align={search[0].value ? 'space-between' : 'end'}>
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/coverage-details/coverage-details.tsx:59]: `search[0]`?
if u need it, use turple

File: src/pages/agent-full-page/coverage-plugin/test-details/test-details.tsx [hunk 3]
```diff
@@ -1,31 +1,37 @@
 import * as React from 'react';
 import { BEM } from '@redneckz/react-bem-helper';
 import {
-  Icons, Panel, Column, Table,
+  Icons, Column, Table, Panel,
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/test-details/test-details.tsx:4]: ?

File: src/pages/agent-full-page/coverage-plugin/test-details/test-details.tsx [hunk 4]
```diff
@@ -1,31 +1,37 @@
...
 } from '@drill4j/ui-kit';
 
 import { capitalize } from 'utils';
 import { AssociatedTests } from 'types/associated-tests';
-import { MethodCoveredByTest } from 'types/method-covered-by-test';
 import { Cells } from 'components';
+import {
+  useTableActionsState,
+} from 'modules/table-actions';
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/test-details/test-details.tsx:12]: ?
```suggestion
} from 'modules';

```

File: src/pages/agent-full-page/coverage-plugin/tests/tests.tsx [hunk 5]
```diff
@@ -1,25 +1,24 @@
 import * as React from 'react';
 
-import { AssociatedTests } from 'types/associated-tests';
-import { MethodCoveredByTest } from 'types/method-covered-by-test';
 import { TestSummary } from 'types/test-summary';
 import { TestTypeSummary } from 'types/test-type-summary';
+import { TableActionsProvider } from 'modules/table-actions';
```
Comment (pavyarov) [inline src/pages/agent-full-page/coverage-plugin/tests/tests.tsx:5]: ?

**Review by pavyarov — CHANGES_REQUESTED** [review]

File: src/modules/table-actions/table-actions-context.ts (hunk 1 above)
Comment (IliaSolovev) [reply src/modules/table-actions/table-actions-context.ts:7]: yes, default value is "EQ"

File: src/modules/table-actions/reducer.ts [hunk 6]
```diff
@@ -1,25 +1,26 @@
...
 export const actionsReducer = (state: TableActionsState, action: Action): TableActionsState => {
   switch (action.type) {
     case SET_SEARCH:
-      return { ...state, search: { ...state.search, value: action.payload } };
+      return { ...state, search: action.payload };
     case TOGGLE_ORDER:
       return {
         ...state,
-        sort: {
+        sort: state.sort.map(({ fieldName, order }) => ({
```
Comment (RomanAchilov) [inline src/modules/table-actions/reducer.ts:20]: const [sort] = state

File: src/types/search.ts [hunk 7]
```diff
@@ -1,4 +1,5 @@
 export interface Search {
   fieldName: string;
   value: string;
+  op?: 'CONTAINS' | 'EQ';
```
Comment (RomanAchilov) [inline src/types/search.ts:4]: type Op = 'CONTAINS' | 'EQ';

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-list.tsx [hunk 8]
```diff
@@ -23,17 +24,18 @@ const testsToRunList = BEM(styles);
...
+  const [query] = search;
   return (
     <div className={className}>
       <TestsToRunHeader agentType={agentType} testsToRunCount={testToRunCount} />
       <Title data-test="tests-to-run-list:title">SUGGESTED TESTS</Title>
       <div>
         <SearchPanel
-          onSearch={(value) => setSearch({ ...search, value })}
-          searchQuery={search.value}
+          onSearch={(value) => setSearch([{ fieldName: 'name', value, op: 'CONTAINS' }])}
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-list.tsx:37]: ?

File: src/hooks/use-build-version.ts [hunk 9]
```diff
@@ -19,16 +19,16 @@ export function useBuildVersion<Data>(topic: string, search?: Search, sort?: Sor
         agentId,
         buildVersion,
         type: 'AGENT',
-        searchStatement: search,
-        sortStatement: sort,
+        filters: filters?.map(({ fieldName, value, op }) => ({ field: fieldName, value, op })),
```
Comment (RomanAchilov) [inline src/hooks/use-build-version.ts:22]: ?

File: src/hooks/use-build-version.ts [hunk 10]
```diff
@@ -19,16 +19,16 @@ export function useBuildVersion<Data>(topic: string, search?: Search, sort?: Sor
         agentId,
         buildVersion,
         type: 'AGENT',
-        searchStatement: search,
-        sortStatement: sort,
+        filters: filters?.map(({ fieldName, value, op }) => ({ field: fieldName, value, op })),
+        orderBy: orderBy?.map(({ fieldName, order }) => ({ field: fieldName, order })),
```
Comment (RomanAchilov) [inline src/hooks/use-build-version.ts:23]: ?

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-list.tsx [hunk 11]
```diff
@@ -23,17 +24,18 @@ const testsToRunList = BEM(styles);
...
+          searchQuery={query.value}
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-list.tsx:38]: searchQuery

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/hooks/use-build-version.ts [hunk 12]
```diff
@@ -5,7 +5,7 @@ import { Search } from 'types/search';
 import { Sort } from 'types/sort';
 import { usePluginState } from 'pages/agent-full-page/store';
 
-export function useBuildVersion<Data>(topic: string, search?: Search, sort?: Sort): Data | null {
+export function useBuildVersion<Data>(topic: string, filters?: Search[], orderBy?: Sort[]): Data | null {
```
Comment (RomanAchilov) [inline src/hooks/use-build-version.ts:8]: filters?: Search[] 
?

File: src/modules/tests-to-run/tests-to-run-list/tests-to-run-list.tsx [hunk 13]
```diff
@@ -10,6 +10,7 @@ import { useBuildVersion } from 'hooks';
 import { CoveredMethodsByTestSidebar } from 'modules';
 import { Metrics } from 'types/metrics';
 import { capitalize } from 'utils';
+import { Search } from 'types/search';
```
Comment (RomanAchilov) [inline src/modules/tests-to-run/tests-to-run-list/tests-to-run-list.tsx:13]: ⬆️

File: src/types/search.ts [hunk 14]
```diff
@@ -1,4 +1,7 @@
+type OP = 'CONTAINS' | 'EQ';
```
Comment (pavyarov) [inline src/types/search.ts:1]: Operator

**Review by pavyarov — CHANGES_REQUESTED** [review]

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

**Pull request merged**